cmd/maya-exporter/app/collector/cstorcollector.go

```diff
@@ -0,0 +1,336 @@
+	}
+}
+
+// gaugeList returns the list of the registered gauge variables
+func (c *CstorStatsExporter) gaugesList() []prometheus.Gauge {
+	return []prometheus.Gauge{
+		c.Metrics.readIOPS,
+		c.Metrics.writeIOPS,
+		c.Metrics.readTimePS,
+		c.Metrics.writeTimePS,
+		c.Metrics.readBlockCountPS,
+		c.Metrics.writeBlockCountPS,
+		c.Metrics.actualUsed,
+		c.Metrics.logicalSize,
+		c.Metrics.sectorSize,
+		c.Metrics.readLatency,
+		c.Metrics.writeLatency,
+		c.Metrics.avgReadBlockCountPS,
+		c.Metrics.avgWriteBlockCountPS,
+		c.Metrics.sizeOfVolume,
+	}
+}
+
+// counterList returns the list of registered counter variables
+func (c *CstorStatsExporter) countersList() []prometheus.Collector {
+	return []prometheus.Collector{
+		c.Metrics.volumeUpTime,
+		c.Metrics.connectionErrorCounter,
+		c.Metrics.connectionRetryCounter,
+	}
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by this Collector to the provided channel and returns once
+// the last descriptor has been sent. The sent descriptors fulfill the
+// consistency and uniqueness requirements described in the Desc
+// documentation. (It is valid if one and the same Collector sends
+// duplicate descriptors. Those duplicates are simply ignored. However,
+// two different Collectors must not send duplicate descriptors.) This
+// method idempotently sends the same descriptors throughout the
+// lifetime of the Collector. If a Collector encounters an error while
+// executing this method, it must send an invalid descriptor (created
+// with NewInvalidDesc) to signal the error to the registry.
+
+// Describe describes all the registered stats metrics from the OpenEBS volumes.
+func (c *CstorStatsExporter) Describe(ch chan<- *prometheus.Desc) {
+	for _, gauge := range c.gaugesList() {
+		gauge.Describe(ch)
+	}
+
+	for _, counter := range c.countersList() {
+		counter.Describe(ch)
+	}
+}
+
+// Collect is called by the Prometheus registry when collecting
+// metrics. The implementation sends each collected metric via the
+// provided channel and returns once the last metric has been sent. The
+// descriptor of each sent metric is one of those returned by
+// Describe. Returned metrics that share the same descriptor must differ
+// in their variable label values. This method may be called
+// way. Bloc	king occurs at the expense of total performance of rendering
+// concurrently and must therefore be implemented in a concurrency safe
+// all registered metrics. Ideally, Collector implementations support
+// concurrent readers.
+
+// Collect collects all the registered stats metrics from the OpenEBS volumes.
+// It tries to reconnect with the volume if there is any error via a goroutine.
+func (c *CstorStatsExporter) Collect(ch chan<- prometheus.Metric) {
+	// no need to catch the error as exporter should work even if
+	// there are failures in collecting the metrics due to connection
+	// issues or anything else.
+	_ = c.collector()
```

Comment:
This method seems to have been written to improve the UT code coverage instead of having the body pasted here. Please mention the same in a comment.